scipy/stats/tests/test_hypotests.py

```diff
@@ -405,3 +406,150 @@ def test_contingency_table_return(self):
         y = np.arange(10)
         res = stats.somersd(x, y)
         assert_equal(res.table, np.eye(10))
+
+
+class TestBarnardExact(object):
+    """Some tests to show that barnard_exact() works correctly."""
+
+    @pytest.mark.parametrize(
+        "input_sample,expected",
+        [
+            ([[43, 40], [10, 39]], (3.555406779643, 0.000362832367)),
+            ([[100, 2], [1000, 5]], (-1.776382925679, 0.135126970878)),
+            ([[2, 7], [8, 2]], (-2.518474945157, 0.019210815430)),
+            ([[5, 1], [10, 10]], (1.449486150679, 0.156277546306)),
+            ([[5, 15], [20, 20]], (-1.851640199545, 0.066363501421)),
+            ([[5, 16], [20, 25]], (-1.609639949352, 0.116984852192)),
+            ([[10, 5], [10, 1]], (-1.449486150679, 0.177536588915)),
+            ([[5, 0], [1, 4]], (2.581988897472, 0.013671875000)),
+            ([[0, 1], [3, 2]], (-1.095445115010, 0.509667991877)),
```

Comment:
Why is the statistic the negative of what R reports?

![image](https://user-images.githubusercontent.com/6570539/110227154-27ab2600-7eaa-11eb-901b-4670a26407d6.png)

Answer: because R's test statistic is the negative of ours - and it's just a convention?

This probably relates to our discussion of the fisher test statistic a long time ago and why `fisher_exact`'s convention is the way it is?

Reply:
Exactly. I wanted to change `fisher_exact` behavior (basically, invert the way fisher_exact compute its statistics) so that `fisher_exact` and `barnard_exact` can have similar behavior and therefore have similar values as we can found in papers and other different softwares as R on internet. That would mean possibly breaking production code for people who are using fisher_exact in older version. 
@jb-leger explain very well the issue in the following [comment](https://github.com/scipy/scipy/pull/13441#discussion_r571200033)

Reply:
Got it!